screens/Directions.js

```diff
@@ -0,0 +1,305 @@
+     */
+    const updateMapRegionToInstruction = (index) => {
+        setTimeout(() => {
+            mapRef.current.fitToCoordinates(
+                destinationResponse.steps[index].polylines,
+                { edgePadding: { bottom: 100, right: 50, left: 50, top: 300 }, animated: true, });
+            
+        }, 100);
+    }
+
+    /**
+     * Description: The map has to be initialized to the first instruction.
+     * Particularity: We used the reference of the mapview used in ref={mapRef}. 
+     */
+    const initMapRegion = () => {
+        setTimeout(() => {
+            mapRef.current.fitToCoordinates(
+                destinationResponse.steps[0].polylines,
+                { edgePadding: { bottom: 100, right: 50, left: 50, top: 300 }, animated: true, });
+        }, 100);
+    }
+    /**
+     * Description: This method will increment the instructionIndex so that we can render the next instruction 
+     * in the MapView. 
+     * How: 
+     * 1. Making sure it's not the last instruction, if it is, we disabled the button by setting 
+     * setIsLastInstruction to true. If it's not, we increment the counter and setIsLastInstruction
+     * to false.
+     * 2. We update the mapView by calling updateMapRegionToInstruction with the index incremented.
+     */
+    const goToNextInstruction = () => {
+        if(instructionIndex >= destinationResponse.steps.length - 1) {
+            setIsLastInstruction(true);
+        } else {
+        setInstructionIndex(instructionIndex + 1);
+        setIsFirstInstruction(false);
+        updateMapRegionToInstruction(instructionIndex + 1);      
+        }
+    }
+    /**
+     * Same as goToNextInstruction but for previous instruction
+     */
+    const goToPreviousInstruction = () => {
+        if(instructionIndex == 0) {
+            setIsFirstInstruction(true);
+        } else {
+        setInstructionIndex(instructionIndex - 1);
+        setIsLastInstruction(false);
+        updateMapRegionToInstruction(instructionIndex - 1);
+        }
+    }
+
+    return (
+        <View>
+            <MapView
+                ref={mapRef}
+                style={styles.map}
+                provider={PROVIDER_GOOGLE}
+                showsUserLocation={true}
+                showsCompass={true}
+                showsBuildings={true}
+                onLayout={initMapRegion}
+                showsIndoors = {false}
+                // customMapStyle = {mapStyleJsonDownloaded} Refer to TODO: A)
+            >
+                <Polyline
+                coordinates = {destinationResponse ? destinationResponse.generalRouteInfo.overviewPolyline : []}
+                strokeWidth = {6}
+                strokeColor = "pink"
+                />
+            </MapView>
+            <View style ={styles.circleCurrentLocation}>
+                <CurrentLocationButton mapReference ={mapRef}/>
+            </View>
+            <View style={styles.navigationHeader}>
+                <View style={{ top: "15%" }}>
```

Comment:
inline styling

Reply:
I really don't like to create styling for 1 attribute, but for inconsistency I get it. Will do.